src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -72,7 +71,7 @@ public GatewayMetaState(Settings settings, NodeEnvironment nodeEnv, MetaStateSer
         if (DiscoveryNode.masterNode(settings) || DiscoveryNode.dataNode(settings)) {
             nodeEnv.ensureAtomicMoveSupported();
         }
-        if (DiscoveryNode.masterNode(settings)) {
+        if (DiscoveryNode.masterNode(settings) || DiscoveryNode.dataNode(settings)) {
```

Comment:
The pre20Upgrade() we should still do because this change is going to 1.x as well. But we probably need not check on data nodes for pre 019 state. Would it harm to keep this check or should I explicitly only check  pre 019 on master nodes and pre 20 on master and data nodes?